### PR TITLE
Introduces the AbstractLayer and Layergroup

### PR DESCRIPTION
**Pull request by KaiVolland**
This introduces the AbstractLayer and Layergroup.

### CODE REVIEW

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/layer/Layergroup.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+ * and an appearance
+ *
+ * @author Kai Volland
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+@Entity
+@Table
+public class Layergroup extends AbstractLayer {
+
```
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/layer/Layergroup.java]: CamelCase? -> `LayerGroup`

**Comment by buehner** [comment]
I just had a very quick look at this, but as `AbstractLayer` is abstract, shouldn't the corresponding DAO/Service/Controller also be abstract (or are even not really necessary). I will have a deeper look at this tomorrow.

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/layer/Layergroup.java (hunk 1 above)
Comment (KaiVolland) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/layer/Layergroup.java]: Isn't Layergroup one word?

**Comment by KaiVolland** [comment]
I need a restinterface where i receive all layers and layergroups which are subclasses of AbstractLayer. I didn't test if i can realise that with abstract service and dao... I'll try that tomorrow.

**Comment by chrismayer** [comment]
Hi @KaiVolland and @buehner,
do you think it will be possible to tag the current master as ``v0.0.4`` before merging this? What do you think? TIA!

**Comment by KaiVolland** [comment]
@chrismayer Good idea. We changed a lot of things since the last version tag. Just do it.

**Comment by buehner** [comment]
> Just do it.

I think that is easier said than done. As far as i know, currently only @dnlkoch und me can release in a way that assures that also the maven repos are updated (by using the release script, which requires some additional configuration like registering a SSH key on github).

But of course we can release `v0.0.4`. I will try to do this within the next hour. We should also document how such a release can be performed.

**Comment by chrismayer** [comment]
Cool, thank you guys!

**Comment by buehner** [comment]
You'll find some background information here: #113

**Comment by buehner** [comment]
The `v0.0.4` release is finished (on the side of github). I think this version will appear at http://mvnrepository.com/artifact/de.terrestris/shogun2/0.0.4 within the next few hours

File: src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/layer/Layergroup.java (hunk 1 above)
Comment (buehner) [inline src/shogun2-core/shogun2-model/src/main/java/de/terrestris/shogun2/model/layer/Layergroup.java]: Yes. In the end this is a matter of taste.

I prefer the CamelCase version as it is better readable. We should also handle this consistently within the project and as we already have a model `UserGroup`, i would also like to have `LayerGroup` here :wink:

**Comment by chrismayer** [comment]
Hi @buehner,
many thanks for the release! I just checked mvnrepository.com and it seems that several artifacts got updated (e.g. [shogun2-service](http://mvnrepository.com/artifact/de.terrestris/shogun2-service)) and some not (e.g. [shogun2](http://mvnrepository.com/artifact/de.terrestris/shogun2/0.0.4)).

Just a matter of time or went something wrong?

**Comment by buehner** [comment]
Hi @chrismayer 

this is really weird. It seems that this problem only occurs on http://mvnrepository.com/

On http://search.maven.org/ everything is released and seems to be fine.

We could not find any abnormalities in the logs, so i think we should wait some more time.

**Comment by KaiVolland** [comment]
@chrismayer @buehner @weskamm Are there any further issues or can i merge this?

**Comment by marcjansen** [comment]
Mvnreposititory.com now has 0.0.4 e.g. for the shogun package. 

http://mvnrepository.com/artifact/de.terrestris/shogun2

We are stil pre-2.0.0-final, and we have a clean way of getting the old behaviour by switching to e.g. 0.0.4. Sooooo:

Please merge.

**Pull request merged**